libsolidity/ast/Types.cpp

```diff
@@ -1155,6 +1157,10 @@ FixedPointType const* RationalNumberType::fixedPointType() const
 		fractionalDigits++;
 	}
 
+	// We need more than 80 digits of precision.
```

Comment:
This is a "breaking change". It leads to `1/3` not being valid anymore.

Reply:
For `1/3`, the value of `fractionalDigits` would be `77`. Currently, this disallows fractions that are less than `(2**256 - 1) / 10**80`

Reply:
Not sure I understand, `1/3` is not finitely representable as a decimal fraction, so `fractionalDigits` would be "infinite" and (at least that was the idea) this code here should refuse it, but maybe an explicit conversion to a fixed point type would be ok, because there you specify the precision.

Reply:
For `1/3`, I think the `while` loop in line 1154 would end after 77 iterations because `value * 10 > maxValue`, when `value` at the beginning, is `1/3`. Because `1/3 * 10**78 > 2**256 - 1`.

Reply:
Oh this is interesting - so it still truncates.

I guess the main question is if we want it to truncate here or not. I guess there should be two functions `mostPreciseFixedPointType' - which is allowed to truncat and `fixedPointType` which is not.

But in general, it's unclear how to handle precision loss. Is the following ok:

```
function f(fixed x) public pure returns (fixed) {
  return x * (1/3);
}
```

Or this one:
```
function f(fixed x) public pure returns (fixed) {
  fixed t = 1/3;
  return x * t;
}
```

Maybe we should always require an explicit cast from a rational number to a fixed point type if there is precision loss?
I think this can be solved by allowing explicit conversions from rational number to fixed point type with precision loss but not implicit conversions.

Reply:
Ah interesting - just re-read ``RationalNumberType::isImplicitlyConvertible`` and ``RationalNumberType::isExplicitlyConvertible`` - and it's implemented in exactly that way already :)